contrib/mesos/pkg/scheduler/podtask/port_mapping.go

```diff
@@ -99,16 +99,16 @@ func wildcardHostPortMapping(t *T, offer *mesos.Offer) ([]HostPortMapping, error
 	remaining := len(wildports)
 	foreachRange(offer, "ports", func(bp, ep uint64) {
 		log.V(3).Infof("Searching for wildcard port in range {%d:%d}", bp, ep)
-		for _, entry := range wildports {
-			if entry.OfferPort != 0 {
+		for i := range wildports {
+			if wildports[i].OfferPort != 0 {
```

Comment:
see https://github.com/mesosphere/kubernetes-mesos/issues/614 and the new unit test case.
in https://github.com/kubernetes/kubernetes/blob/17defc7383914c1f52d30ab4a46e0992cbfc2bae/contrib/mesos/pkg/scheduler/podtask/port_mapping.go#L110 , the copy of the map value is being updated and in above line, it is expecting to hold that value in next outer loop iteration.